src/zope/app/applicationcontrol/zopeversion.py

```diff
@@ -52,60 +44,13 @@ def getZopeVersion(self):
 
         self.result = "Development/Unknown"
 
-        # is this a SVN checkout?
-        svndir = os.path.join(self.path, ".svn")
-        if os.path.isdir(svndir):
-            self.__setSVNVersion(svndir)
-        else:
-            # try to get official Zope release information
-            versionfile = os.path.join(self.path, "version.txt")
-            if os.path.isfile(versionfile):
-                f = file(versionfile)
-                try:
-                    self.result = f.readline().strip() or self.result
-                finally:
-                    f.close()
-        return self.result
+        # try to get official Zope release information
+        versionfile = os.path.join(self.path, "version.txt")
+        if os.path.isfile(versionfile):
+            with open(versionfile) as f:
+                self.result = f.readline().strip() or self.result
```

Comment:
I agree, it's pretty meaningless in most cases. There's a big comment to that effect up in `__init__`, which winds up setting the result to `Meaningless` so this code never even runs in production.

But hey, there were test cases for it, and I suppose some code somewhere could explicitly be setting a path.

I'll open an issue for doing something better in the no-path-given case (but probably won't get to it for a long time personally.)